lua/nvim-treesitter/query_predicates.lua

```diff
@@ -154,6 +154,10 @@ query.add_directive("exclude_children!", function(match, _pattern, _bufnr, pred,
   metadata.content = ranges
 end)
 
+query.add_directive("set-substring-trim!", function(match, _, bufnr, pred, metadata)
```

Comment:
Isn't this just the sane as `offset!`?

Reply:
yep this is basically the same as `offset!` + `trim!`, but that combination wouldn't work without https://github.com/neovim/neovim/pull/19460

Reply:
Then let's make that work first.